### PR TITLE
Integrate runtime with mongodb and add utilities for sub-processes 

### PR DESCRIPTION
**Pull request by BlakeASmith**

### CODE REVIEW

**Comment by igboyes** [comment]
@BlakeASmith There are dependencies being lost when installing `virtool-core`. See Drone logs. Please take a look.

https://cloud.drone.io/virtool/virtool-workflow/14/1/2

**Comment by BlakeASmith** [comment]
> @BlakeASmith There are dependencies being lost when installing `virtool-core`. See Drone logs. Please take a look.
> 
> https://cloud.drone.io/virtool/virtool-workflow/14/1/2

The appropriate dependencies need to be added to `install_requires` in the setup.py for virtool-core. I will submit a PR 
to virtool/virtool-core to fix it

**Review by igboyes — APPROVED** [review]
Please see my comments and address in further work. I am not going to comb through this too hard since a lot of it looks like WIP.

Thanks.

File: tests/conftest.py [hunk 1]
```diff
@@ -1,8 +1,6 @@
 import sys
 from pathlib import Path
 
-
 sys.path.append(str(Path(__file__).parent.parent))
```
[the remark this answers is not in the document]
Comment (BlakeASmith) [reply tests/conftest.py:4]: I was having some issues with importing `virtool_workflow` from the tests. This was the fix at the time, just to add the root project directory to the system path. 

It seems that the tests run fine without it now, so It'll be removed in the next PR

**Pull request merged**